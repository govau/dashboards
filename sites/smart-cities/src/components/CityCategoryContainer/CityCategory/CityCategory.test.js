import React from 'react';
import { shallow } from 'enzyme';
import CityCategory from './CityCategory';

jest.mock('../../PageBanner/PageBanner', () => 'PageBanner');
jest.mock('../../SubCategorySummary/SubCategorySummary', () => 'SubCategorySummary');
jest.mock('../../../helpers/getSubCategoryOfHeroIndicator', () => () => 'mock sub category');

const defaultProps = {
  category: {
    name: 'Mock category',
    description: 'The description of the category',
    colorName: 'jobs',
    heroIndicatorId: 'population',
    id: 'mock',
    subCategories: [
      {
        name: 'Buying houses',
        colorVariation: '100',
        highlightColorDark: 'HOUSING_600',
        summaryIndicatorIds: [],
      },
      {
        name: 'Living in houses',
        colorVariation: '050',
        highlightColorDark: 'HOUSING_700',
        summaryIndicatorIds: [],
      },
    ],
  },
  city: {
    name: 'Perth',
    id: 'perth',
    indicators: {},
  },
  cities: [
    {
      id: 'perth',
      description: 'description of Perth',
      name: 'Perth',
      indicators: {}
    },
    {
      id: 'sydney',
      description: 'description of Sydney',
      name: 'Sydney',
      indicators: {}
    },
    {
      id: 'melbourne',
      description: 'description of Melbourne',
      name: 'Melbourne',
      indicators: {}
    },
  ],
  toggleCitySelected: () => {},
  setShowChartPatterns: () => {},
  showChartPatterns: true,
};

it('should match Snapshot', () => {
  const component = shallow(
    <CityCategory {...defaultProps} />
  );

  expect(component.debug()).toMatchSnapshot();
});

it('should pass the sub category of the hero indicator to the banner', () => {
  const component = shallow(
    <CityCategory {...defaultProps} />
  );

  expect(component.find('PageBanner').prop('indicatorSubCategory')).toBe('mock sub category');
});
