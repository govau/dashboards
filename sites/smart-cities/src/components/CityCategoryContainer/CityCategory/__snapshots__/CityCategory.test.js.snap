// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match Snapshot 1`] = `
"<PageWrapper cityId=\\"perth\\" categoryId=\\"mock\\" cityName=\\"Perth\\" categoryColorName=\\"jobs\\">
  <PageBanner colorName=\\"jobs\\" description=\\"The description of the category\\" indicatorId=\\"population\\" title=\\"Mock category\\" cities={{...}} city={{...}} indicatorSubCategory={{...}} />
  <SubCategorySummary name=\\"Buying houses\\" colorVariation=\\"100\\" highlightColorDark=\\"HOUSING_600\\" summaryIndicatorIds={{...}} categoryColorName=\\"jobs\\" city={{...}} />
  <SubCategorySummary name=\\"Living in houses\\" colorVariation=\\"050\\" highlightColorDark=\\"HOUSING_700\\" summaryIndicatorIds={{...}} categoryColorName=\\"jobs\\" city={{...}} />
  <CompareCities category={{...}} city={{...}} cities={{...}} toggleCitySelected={[Function]} />
  <SubCategoryCharts subCategory={{...}} colorName=\\"jobs\\" cities={{...}} city={{...}} heroIndicatorId=\\"population\\" highlightColorDark=\\"HOUSING_600\\" setShowChartPatterns={[Function]} showChartPatterns={true} />
  <SubCategoryCharts subCategory={{...}} colorName=\\"jobs\\" cities={{...}} city={{...}} heroIndicatorId=\\"population\\" highlightColorDark=\\"HOUSING_700\\" setShowChartPatterns={[Function]} showChartPatterns={true} />
</PageWrapper>"
`;
